update_ps_doc.py

```diff
@@ -0,0 +1,76 @@
+from collections import defaultdict
+import os
+import re
+
+import pybtex
+
+import pybamm
+from pybamm.parameters import parameter_sets
+
+
+DOC_INTRO = """
+Parameter sets from papers. The 'citation' entry provides a reference to the appropriate
+paper in the file "pybamm/CITATIONS.txt". To see which parameter sets have been used in
+your simulation, add the line "pybamm.print_citations()" to your script.
+"""
+
+
+def main():
+    parameter_set_dict = defaultdict(list)
+    for ps_name, ps_dict in parameter_sets.__dict__.items():
+        if not isinstance(ps_dict, dict):
+            continue
+        elif "citation" not in ps_dict or "chemistry" not in ps_dict:
+            continue
+
+        chemistry = ps_dict["chemistry"]
+        citation = ps_dict["citation"]
+
+        # Enclose citation in a list if not already enclosed
+        if not isinstance(citation, list):
+            citation = [citation]
+
+        parameter_set_dict[chemistry].append((ps_name, citation))
+
+    output_list = [DOC_INTRO]
+    citations_file = os.path.join(pybamm.root_dir(), "pybamm", "CITATIONS.txt")
+
+    for ps_chemistry in sorted(parameter_set_dict.keys()):
+        ps_citations = parameter_set_dict[ps_chemistry]
+        chem_name = ps_chemistry.capitalize() + " " + "parameter sets"
+        output_list.append(chem_name)
+        dashes = "-" * len(ps_chemistry) + "-" * 15
+        output_list.append(dashes)
+
+        for ps_name, ps_citation in sorted(ps_citations):
+            citations = pybtex.format_from_file(
+                citations_file,
+                style="plain",
+                output_backend="plaintext",
+                citations=ps_citation,
+                nocite=True,
+            )
+            # Remove citation labels "[3]"
+            citations = re.sub(r"(?:^|\n)(\[\d+\]\s)", "", citations)
+            # Break line at the first space before 80 characters
+            citations = re.findall(r"(.{1,80})(?:\s|$)", citations)
+            citations = "\n".join(map(lambda x: " " * 8 + x, citations))
+            ps_doc = f"    * {ps_name:} :\n{citations}"
+            output_list.append(ps_doc)
+
+    output = "\n".join(output_list)
+    output += "\n"
+
+    with open(
+        os.path.join(pybamm.root_dir(), "pybamm", "parameters", "parameter_sets.py"),
+        "r+",
```

Comment:
I tried to run the script in any folder, but as soon as I place the script in any folder, it stops working. To debug it I printed the path of the changed file and I got `/home/priyanshu/anaconda3/envs/pybamm/lib/python3.8/site-packages/pybamm/parameters/parameter_sets.py`, it should have used the files from my local checkout.
Maybe it has something to do with the path or maybe I should not install `pip install pybamm`  in the workflow, the workflow should use the local checkout.

Reply:
Hi @priyanshuone6 , `pip install pybamm` installs pybamm from pypi.org, to your 
Maybe give `pip install --editable .` a try. 

- `pip install .` installs pybamm from the `setup.py` located in the current directory
- `--editable` redirects your `site-packages` directory to your development PyBaMM directory (you called the pip command from)

Reply:
Thanks, it worked!
We should add `pip install --editable .` under the contributing guidelines as not everyone uses virtual env created using `.tox/dev`. I personally like using conda virtual envs and it would be convenient to list an additional option to cover all those who might want to use any other virtual environment.

Reply:
Yes - a lot of people have different preferred method to manage their projects.  We expect a significant part of PyBaMM's new contributors not to be familiar with the python ecosystem, particularly managing virtual environments. Using tox might make it easier. If you prefer using another method, you're welcome to do so (I personally do so). More detailed installation instructions for PyBaMM contributors are available at https://pybamm.readthedocs.io/en/latest/install/install-from-source.html

Reply:
Thanks for letting me know 😀